In [104]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import glob

### Загрузка датасетов

In [105]:
data_CA = pd.read_csv('data\\CAvideos.csv', encoding = 'KOI8-R')
data_DE = pd.read_csv('data\\DEvideos.csv', encoding = 'KOI8-R')
data_FR = pd.read_csv('data\\FRvideos.csv', encoding = 'KOI8-R')
data_GB = pd.read_csv('data\\GBvideos.csv', encoding = 'KOI8-R')
data_IN = pd.read_csv('data\\INvideos.csv', encoding = 'KOI8-R')
data_JP = pd.read_csv('data\\JPvideos.csv', encoding = 'KOI8-R')
data_KR = pd.read_csv('data\\KRvideos.csv', encoding = 'KOI8-R')
data_MX = pd.read_csv('data\\MXvideos.csv', encoding = 'KOI8-R')
data_RU = pd.read_csv('data\\RUvideos.csv', encoding = 'KOI8-R')
data_US = pd.read_csv('data\\USvideos.csv', encoding = 'KOI8-R')

Общий датасет

In [106]:
paths = glob.glob('data/*.csv', recursive = True)
sorted(paths)

['data\\CAvideos.csv',
 'data\\DEvideos.csv',
 'data\\FRvideos.csv',
 'data\\GBvideos.csv',
 'data\\INvideos.csv',
 'data\\JPvideos.csv',
 'data\\KRvideos.csv',
 'data\\MXvideos.csv',
 'data\\RUvideos.csv',
 'data\\USvideos.csv']

In [107]:
data = pd.concat([pd.read_csv(i, encoding = 'KOI8-R') for i in paths])

In [108]:
data.head()

video_id trending_date  \
0  n1WpP7iowLc      17.14.11   
1  0dBIkQ4Mz1M      17.14.11   
2  5qpjK5DgCt4      17.14.11   
3  d380meD0W0M      17.14.11   
4  2Vv-BfVoq4g      17.14.11   

                                               title channel_title  \
0        Eminem - Walk On Water (Audio) ft. Beyoncц╘    EminemVEVO   
1                      PLUSH - Bad Unboxing Fan Mail     iDubbbzTV   
2  Racist Superman | Rudy Mancuso, King Bach & Le...  Rudy Mancuso   
3                           I Dare You: GOING BALD!?      nigahiga   
4        Ed Sheeran - Perfect (Official Music Video)    Ed Sheeran   

   category_id              publish_time  \
0           10  2017-11-10T17:00:03.000Z   
1           23  2017-11-13T17:00:00.000Z   
2           23  2017-11-12T19:05:24.000Z   
3           24  2017-11-12T18:01:41.000Z   
4           10  2017-11-09T11:04:14.000Z   

                                                tags     views    likes  \
0  Eminem|"Walk"|"On"|"Water"|"Aftermath/Shady/In...  17158579   787425   
1  plush|"bad unboxing"|"unboxing"|"fan mail"|"id...   1014651   127794   
2  racist superman|"rudy"|"mancuso"|"king"|"bach"...   3191434   146035   
3  ryan|"higa"|"higatv"|"nigahiga"|"i dare you"|"...   2095828   132239   
4  edsheeran|"ed sheeran"|"acoustic"|"live"|"cove...  33523622  1634130   

   dislikes  comment_count                                  thumbnail_link  \
0     43420         125882  https://i.ytimg.com/vi/n1WpP7iowLc/default.jpg   
1      1688          13030  https://i.ytimg.com/vi/0dBIkQ4Mz1M/default.jpg   
2      5339           8181  https://i.ytimg.com/vi/5qpjK5DgCt4/default.jpg   
3      1989          17518  https://i.ytimg.com/vi/d380meD0W0M/default.jpg   
4     21082          85067  https://i.ytimg.com/vi/2Vv-BfVoq4g/default.jpg   

   comments_disabled  ratings_disabled  video_error_or_removed  \
0              False             False                   False   
1              False             False                   False   
2              False             False                   False   
3              False             False                   False   
4              False             False                   False   

                                         description  
0  Eminem's new track Walk on Water ft. Beyoncц╘ ...  
1  STill got a lot of packages. Probably will las...  
2  WATCH MY PREVIOUS VIDEO Б√╤ \n\nSUBSCRIBE Б√╨ ...  
3  I know it's been a while since we did this sho...  
4  П÷▌╖: https://ad.gt/yt-perfect\nП÷▓╟: https://...

In [109]:
data.shape

(375942, 16)

In [110]:
data.dtypes

video_id                  object
trending_date             object
title                     object
channel_title             object
category_id                int64
publish_time              object
tags                      object
views                      int64
likes                      int64
dislikes                   int64
comment_count              int64
thumbnail_link            object
comments_disabled           bool
ratings_disabled            bool
video_error_or_removed      bool
description               object
dtype: object

In [111]:
data.isna().sum()

video_id                      0
trending_date                 0
title                         0
channel_title                 0
category_id                   0
publish_time                  0
tags                          0
views                         0
likes                         0
dislikes                      0
comment_count                 0
thumbnail_link                0
comments_disabled             0
ratings_disabled              0
video_error_or_removed        0
description               19478
dtype: int64

Из 375942 данных 19478 видео не имеют описания, что составляет 5%. Предпологаем, что это не ошибка, а в видео действительно нет описания. В дальнейшем исследуем зависимость популярности видео от наличия описания.

**Немного модифицируем таблицу.** Пофиксим формат времени, разделим информацию с *publishing_time* на *publishing_time* и *publishing_date*. Также из json файлов получаем информацию о жанре видео.

In [112]:
data['trending_date'] = pd.to_datetime(data['trending_date'], format='%y.%d.%m')
data['publish_time'] = pd.to_datetime(data['publish_time'], format ='%Y-%m-%dT%H:%M:%S.%fZ')
data.head()

video_id trending_date  \
0  n1WpP7iowLc    2017-11-14   
1  0dBIkQ4Mz1M    2017-11-14   
2  5qpjK5DgCt4    2017-11-14   
3  d380meD0W0M    2017-11-14   
4  2Vv-BfVoq4g    2017-11-14   

                                               title channel_title  \
0        Eminem - Walk On Water (Audio) ft. Beyoncц╘    EminemVEVO   
1                      PLUSH - Bad Unboxing Fan Mail     iDubbbzTV   
2  Racist Superman | Rudy Mancuso, King Bach & Le...  Rudy Mancuso   
3                           I Dare You: GOING BALD!?      nigahiga   
4        Ed Sheeran - Perfect (Official Music Video)    Ed Sheeran   

   category_id        publish_time  \
0           10 2017-11-10 17:00:03   
1           23 2017-11-13 17:00:00   
2           23 2017-11-12 19:05:24   
3           24 2017-11-12 18:01:41   
4           10 2017-11-09 11:04:14   

                                                tags     views    likes  \
0  Eminem|"Walk"|"On"|"Water"|"Aftermath/Shady/In...  17158579   787425   
1  plush|"bad unboxing"|"unboxing"|"fan mail"|"id...   1014651   127794   
2  racist superman|"rudy"|"mancuso"|"king"|"bach"...   3191434   146035   
3  ryan|"higa"|"higatv"|"nigahiga"|"i dare you"|"...   2095828   132239   
4  edsheeran|"ed sheeran"|"acoustic"|"live"|"cove...  33523622  1634130   

   dislikes  comment_count                                  thumbnail_link  \
0     43420         125882  https://i.ytimg.com/vi/n1WpP7iowLc/default.jpg   
1      1688          13030  https://i.ytimg.com/vi/0dBIkQ4Mz1M/default.jpg   
2      5339           8181  https://i.ytimg.com/vi/5qpjK5DgCt4/default.jpg   
3      1989          17518  https://i.ytimg.com/vi/d380meD0W0M/default.jpg   
4     21082          85067  https://i.ytimg.com/vi/2Vv-BfVoq4g/default.jpg   

   comments_disabled  ratings_disabled  video_error_or_removed  \
0              False             False                   False   
1              False             False                   False   
2              False             False                   False   
3              False             False                   False   
4              False             False                   False   

                                         description  
0  Eminem's new track Walk on Water ft. Beyoncц╘ ...  
1  STill got a lot of packages. Probably will las...  
2  WATCH MY PREVIOUS VIDEO Б√╤ \n\nSUBSCRIBE Б√╨ ...  
3  I know it's been a while since we did this sho...  
4  П÷▌╖: https://ad.gt/yt-perfect\nП÷▓╟: https://...

In [113]:
data.insert(5, 'publish_date', data['publish_time'].dt.date)
data['publish_time'] = data['publish_time'].dt.time

In [114]:
data.head()

video_id trending_date  \
0  n1WpP7iowLc    2017-11-14   
1  0dBIkQ4Mz1M    2017-11-14   
2  5qpjK5DgCt4    2017-11-14   
3  d380meD0W0M    2017-11-14   
4  2Vv-BfVoq4g    2017-11-14   

                                               title channel_title  \
0        Eminem - Walk On Water (Audio) ft. Beyoncц╘    EminemVEVO   
1                      PLUSH - Bad Unboxing Fan Mail     iDubbbzTV   
2  Racist Superman | Rudy Mancuso, King Bach & Le...  Rudy Mancuso   
3                           I Dare You: GOING BALD!?      nigahiga   
4        Ed Sheeran - Perfect (Official Music Video)    Ed Sheeran   

   category_id publish_date publish_time  \
0           10   2017-11-10     17:00:03   
1           23   2017-11-13     17:00:00   
2           23   2017-11-12     19:05:24   
3           24   2017-11-12     18:01:41   
4           10   2017-11-09     11:04:14   

                                                tags     views    likes  \
0  Eminem|"Walk"|"On"|"Water"|"Aftermath/Shady/In...  17158579   787425   
1  plush|"bad unboxing"|"unboxing"|"fan mail"|"id...   1014651   127794   
2  racist superman|"rudy"|"mancuso"|"king"|"bach"...   3191434   146035   
3  ryan|"higa"|"higatv"|"nigahiga"|"i dare you"|"...   2095828   132239   
4  edsheeran|"ed sheeran"|"acoustic"|"live"|"cove...  33523622  1634130   

   dislikes  comment_count                                  thumbnail_link  \
0     43420         125882  https://i.ytimg.com/vi/n1WpP7iowLc/default.jpg   
1      1688          13030  https://i.ytimg.com/vi/0dBIkQ4Mz1M/default.jpg   
2      5339           8181  https://i.ytimg.com/vi/5qpjK5DgCt4/default.jpg   
3      1989          17518  https://i.ytimg.com/vi/d380meD0W0M/default.jpg   
4     21082          85067  https://i.ytimg.com/vi/2Vv-BfVoq4g/default.jpg   

   comments_disabled  ratings_disabled  video_error_or_removed  \
0              False             False                   False   
1              False             False                   False   
2              False             False                   False   
3              False             False                   False   
4              False             False                   False   

                                         description  
0  Eminem's new track Walk on Water ft. Beyoncц╘ ...  
1  STill got a lot of packages. Probably will las...  
2  WATCH MY PREVIOUS VIDEO Б√╤ \n\nSUBSCRIBE Б√╨ ...  
3  I know it's been a while since we did this sho...  
4  П÷▌╖: https://ad.gt/yt-perfect\nП÷▓╟: https://...

In [115]:
data.dtypes

video_id                          object
trending_date             datetime64[ns]
title                             object
channel_title                     object
category_id                        int64
publish_date                      object
publish_time                      object
tags                              object
views                              int64
likes                              int64
dislikes                           int64
comment_count                      int64
thumbnail_link                    object
comments_disabled                   bool
ratings_disabled                    bool
video_error_or_removed              bool
description                       object
dtype: object

In [116]:
import json

id_dict = {}

with open('data/US_category_id.json', 'r') as JSON:
    json_dict = json.load(JSON)
    for fields in json_dict['items']:
        id_dict[fields['id']] = fields['snippet']['title']
            
id_dict

{'1': 'Film & Animation',
 '2': 'Autos & Vehicles',
 '10': 'Music',
 '15': 'Pets & Animals',
 '17': 'Sports',
 '18': 'Short Movies',
 '19': 'Travel & Events',
 '20': 'Gaming',
 '21': 'Videoblogging',
 '22': 'People & Blogs',
 '23': 'Comedy',
 '24': 'Entertainment',
 '25': 'News & Politics',
 '26': 'Howto & Style',
 '27': 'Education',
 '28': 'Science & Technology',
 '29': 'Nonprofits & Activism',
 '30': 'Movies',
 '31': 'Anime/Animation',
 '32': 'Action/Adventure',
 '33': 'Classics',
 '34': 'Comedy',
 '35': 'Documentary',
 '36': 'Drama',
 '37': 'Family',
 '38': 'Foreign',
 '39': 'Horror',
 '40': 'Sci-Fi/Fantasy',
 '41': 'Thriller',
 '42': 'Shorts',
 '43': 'Shows',
 '44': 'Trailers'}

In [117]:
data.insert(len(data.columns), 'genre', pd.Series(id_dict[str(i)] for i in data['category_id']))

In [118]:
data.shape

(375942, 18)